src/Microsoft.OData.Edm/Validation/EdmError.cs

```diff
@@ -48,12 +66,26 @@ public EdmError(EdmLocation errorLocation, EdmErrorCode errorCode, string errorM
         /// <returns>A string representation of the error.</returns>
         public override string ToString()
         {
+            string baseString = Convert.ToString(this.ErrorCode, CultureInfo.InvariantCulture) + " : " + this.ErrorMessage;
+            string errorLevel = Convert.ToString(this.Severity, CultureInfo.InvariantCulture);
+
+            if (this.Severity != Severity.Undefined)
+            {
+
+                if (this.ErrorLocation != null && !(this.ErrorLocation is ObjectLocation))
+                {
+                    return baseString + " : " + errorLevel;
+                }
+
+                return baseString + " : " + errorLevel;

```

Comment:
The if statement seems unnecessary, no?

Reply:
The ToString for severity other than Undefined, error location is no longer given...
Probably need to add test cases to test this method.

Reply:
Resolved